pkg/scheduler/api/unschedule_info.go

```diff
@@ -105,14 +105,33 @@ func NewFitError(task *TaskInfo, node *NodeInfo, message ...string) *FitError {
 		taskName:      task.Name,
 		taskNamespace: task.Namespace,
 		NodeName:      node.Name,
-		Reasons:       message,
 	}
+	sts := make([]*Status, 0, len(message))
+	for _, msg := range message {
+		sts = append(sts, &Status{Reason: msg, Code: Error})
+	}
+	fe.Status = StatusSets(sts)
 	return fe
 }
 
+func NewFitStatus(task *TaskInfo, node *NodeInfo, sts StatusSets) *FitError {
```

Comment:
function name contains `status` but return a Fiteror，this iswired and not descriptive.

Reply:
Done.